aws/resource_aws_redshift_snapshot_schedule_association.go

```diff
@@ -0,0 +1,242 @@
+		d.SetId("")
+		return nil
+	}
+	if isAWSErr(err, redshift.ErrCodeSnapshotScheduleNotFoundFault, "") {
+		log.Printf("[WARN] Redshift Snapshot Schedule (%s) not found, removing from state", scheduleIdentifier)
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error disassociate Redshift Cluster (%s) and Snapshot Schedule (%s) Association: %s", clusterIdentifier, scheduleIdentifier, err)
+	}
+
+	if err := waitForRedshiftSnapshotScheduleAssociationDestroy(conn, 75*time.Minute, clusterIdentifier, scheduleIdentifier); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsRedshiftSnapshotScheduleAssociationParseId(id string) (clusterIdentifier, scheduleIdentifier string, err error) {
+	parts := strings.SplitN(id, "/", 2)
+	if len(parts) != 2 {
+		err = fmt.Errorf("aws_redshift_snapshot_schedule_association id must be of the form <ClusterIdentifier>/<ScheduleIdentifier>")
+		return
+	}
+
+	clusterIdentifier = parts[0]
+	scheduleIdentifier = parts[1]
+	return
+}
+
+func resourceAwsRedshiftSnapshotScheduleAssociationStateRefreshFunc(clusterIdentifier, scheduleIdentifier string, conn *redshift.Redshift) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		log.Printf("[INFO] Reading Redshift Cluster (%s) Snapshot Schedule (%s) Association Information", clusterIdentifier, scheduleIdentifier)
+		resp, err := conn.DescribeSnapshotSchedules(&redshift.DescribeSnapshotSchedulesInput{
+			ClusterIdentifier:  aws.String(clusterIdentifier),
+			ScheduleIdentifier: aws.String(scheduleIdentifier),
+		})
+		if isAWSErr(err, redshift.ErrCodeClusterNotFoundFault, "") {
+			return 42, "destroyed", nil
+		}
+		if isAWSErr(err, redshift.ErrCodeSnapshotScheduleNotFoundFault, "") {
+			return 42, "destroyed", nil
+		}
+		if err != nil {
+			log.Printf("[WARN] Error on retrieving Redshift Cluster (%s) Snapshot Schedule (%s) Association when waiting: %s", clusterIdentifier, scheduleIdentifier, err)
+			return nil, "", err
+		}
+
+		var rcas *redshift.ClusterAssociatedToSchedule
+
+		for _, s := range resp.SnapshotSchedules {
+			if aws.StringValue(s.ScheduleIdentifier) == scheduleIdentifier {
+				for _, c := range s.AssociatedClusters {
+					if aws.StringValue(c.ClusterIdentifier) == clusterIdentifier {
+						rcas = c
+					}
+				}
+			}
+		}
+
+		if rcas == nil {
+			return 42, "destroyed", nil
+		}
+
+		if rcas.ScheduleAssociationState != nil {
+			log.Printf("[DEBUG] Redshift Cluster (%s) Snapshot Schedule (%s) Association status: %s", clusterIdentifier, scheduleIdentifier, aws.StringValue(rcas.ScheduleAssociationState))
+		}
+
+		return rcas, aws.StringValue(rcas.ScheduleAssociationState), nil
+	}
+}
+
+func waitForRedshiftSnapshotScheduleAssociationActive(conn *redshift.Redshift, timeout time.Duration, clusterIdentifier, scheduleIdentifier string) error {
+	// sleep 30 seconds to give it time, until association information included to describe response
+	time.Sleep(30 * time.Second)
```

Comment:
[`resource.StateChangeConf`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#StateChangeConf) accepts a delay in its configuration so this can be moved there 😄 

```go
stateConf := &resource.StateChangeConf{
  // ...
  Delay: 30 * time.Second,
}
```

Reply:
Thanks, fix it.